GoogleUtilities.podspec

```diff
@@ -76,6 +76,13 @@ other Google CocoaPods. They're not intended for direct public usage.
     adss.dependency 'GoogleUtilities/Environment'
   end
 
+  s.subspec 'ComponentContainer' do |cc|
+    cc.source_files = 'GoogleUtilities/ComponentContainer/**/*.[mh]', 'GoogleUtilities/Common/*.h'
+    cc.public_header_files = 'GoogleUtilities/ComponentContainer/Private/*.h'
```

Comment:
Did you mean Public here? This seems like a mistake.

The headers below are also added in a Private directory which seems off. Just by their location it seems like they shouldn't be used (even though you're actually marking them part of public_header_files).

Reply:
@wilhuff This is correct.  From the non-intuitive podspec docs:

private_header_filesmulti-platform
A list of file patterns that should be used to mark private headers.

These patterns are matched against the public headers (or all the headers if no public headers have been specified) to exclude those headers which should not be exposed to the user project and which should not be used to generate the documentation. When the library is built, these headers will appear in the build directory.

Reply:
If the point of this is to make it such that other projects can use the container wouldn't these have to be public headers though?

Reply:
Generally the policy has been to use private headers for APIs that are intended for other Firebase pods that are not intended to be used outside of Firebase.

This is necessary for pods that have both public and internal dependencies.  We could revisit that for pods that are intended purely for internal use and instead use public headers for them.

That brings up another question. I'm not sure we should add another subspec to GoogleUtilities versus making a separate GoogleUtiilitiesComponents pod.

Reply:
First sentence of description: "This will allow non-Firebase SDKs to take advantage of the component system". Private headers won't work in that circumstance.

Reply:
As part of moving this externally to its own CocoaPod, I've made the necessary headers public.